### ISSUE TITLE
my training mistakes

### ISSUE THREAD

**Issue by NieHw** [mpatacchiola/deepgaze#66]
Hello，I used AFLW dataset to train the network. With the epoch, loss would decrease, but  validation accuracy did not improve. How to fix it? Thanks for your apply.

**Comment by mpatacchiola** [comment]
Hi @NieHw

Your model is overfitting the dataset. You need to augment your training set or use a regularization method such as dropout.

**Comment by NieHw** [comment]
> Hi @NieHw
> 
> Your model is overfitting the dataset. You need to augment your training set or use a regularization method such as dropout.

Sorry to bother you again. I noticed that, in 'ex_aflw_parser.py' you use ' label[] /= (np.pi/2) ' to range the labels to (-1, 1), and in the training script, use ’ roll_prediction *= 25... ‘ to recovery the real value of angle. Is there anything wrong with that? 
And when it compute the RMSE, should I use the value of real angle or normalization? 
Thanks for your apply.

**Comment by mpatacchiola** [comment]
Hi @NieHw yes the conversion is not correct. This is how it should be done:

```python
#Min-Max scaling of the label
#Roll
label[:,0] /= ((5.0/36.0)*numpy.pi) #5/36*pi radians correspond to 25 degree
#Pitch
label[:,1] /= (numpy.pi/4.0) #pi/4 radians are 45 degree
#Yaw
label[:,2] /= (numpy.pi/2.0) #pi/2 radians correspond to 90 degree
```
I updated the file `ex_aflw_parser.py`. You must check that the images given as input to the script have been correctly sampled. For instance all the faces with roll>25 and roll<-25 degrees must be removed. The same should be done for the pitch and yaw. Most of the images are included in those ranges, so not much is actually discarded.

**Comment by NieHw** [comment]
> Hi @NieHw yes the conversion is not correct. This is how it should be done:
> 
> ```python
> #Min-Max scaling of the label
> #Roll
> label[:,0] /= ((5.0/36.0)*numpy.pi) #5/36*pi radians correspond to 25 degree
> #Pitch
> label[:,1] /= (numpy.pi/4.0) #pi/4 radians are 45 degree
> #Yaw
> label[:,2] /= (numpy.pi/2.0) #pi/2 radians correspond to 90 degree
> ```
> 
> I updated the file `ex_aflw_parser.py`. You must check that the images given as input to the script have been correctly sampled. For instance all the faces with roll>25 and roll<-25 degrees must be removed. The same should be done for the pitch and yaw. Most of the images are included in those ranges, so not much is actually discarded.

Sorry to bother you again. Did you train three models for each angle？ Or train a model for three angles？

**Comment by mpatacchiola** [comment]
In the paper I trained a CNN for each degree of freedom. In this example the DNN is trained on the three angles all at once. The performance is higher with specialized networks, but the disadvantage is that you have to run all of them in case the estimation of all the angles is required.

**Comment by kamathhrishi** [comment]
Wasn't sure how to contact you @mpatacchiola . The link on the Readme wasn't working. I would be interested in contributing to DeepGaze. Could you give me some lead on how I should be going about it?
